python/paddle/fluid/reader.py

```diff
@@ -27,16 +27,90 @@
 from .dataset import DatasetBase, InMemoryDataset
 
 ### Dygraph DataLoader configs ###
+import atexit
+import os
 import multiprocessing
 import signal
 # NOTE: queue has a different name in python2 and python3
 if sys.version_info[0] == 2:
     import Queue as queue
 else:
     import queue
-# NOTE: [ avoid hanging ] These value is used in getting data from another process
-QUEUE_GET_TIMEOUT = 5
-MAX_GET_FAILED_TIME = 12
+# NOTE: [ avoid hanging & failed quickly ] These value is used in getting data from another process
+QUEUE_GET_TIMEOUT = 60
+
+# NOTE: [ mmap files clear ] If there is still data in the multiprocess queue when the main process finishes reading,
+# the data in the queue needs to be popped. Then the LoDTensor read by the main process
+# from the child process will automatically clear the memory-mapped file.
+multiprocess_queue_set = set()
+
+_registered_exit_funcs = set()
+_executed_exit_funcs = set()
```

Comment:
What is `_executed_exit_funcs` used for?

Reply:
Avoid the same function being executed repeatedly